src/components/dashboard/index.js

```diff
@@ -0,0 +1,335 @@
+                    rel="noopener noreferrer"
+                    color="primary"
+                >
+                    {getMessage("open")}
+                </Button>
+            </CardActions>
+        </Card>
+    );
+};
+
+const DashboardSection = ({ name, kind, items }) => {
+    const classes = useStyles();
+
+    return (
+        <div className={classes.section}>
+            <Typography
+                noWrap
+                gutterBottom
+                variant="h5"
+                component="h5"
+                color="primary"
+            >
+                {name}
+            </Typography>
+
+            <Divider classes={{ root: classes.dividerRoot }} />
+
+            <div className={classes.sectionItems}>
+                {items.map((item) => (
+                    <DashboardItem kind={kind} content={item} key={item.id} />
+                ))}
+            </div>
+        </div>
+    );
+};
+
+const Dashboard = () => {
+    const classes = useStyles();
+    const [data, setData] = useState({});
+
+    useEffect(() => {
+        const fetcher = async () => {
+            let data = await fetch(
+                `/api/dashboard?limit=${constants.SECTION_ITEM_LIMIT}`,
+                {
+                    method: "GET",
+                    cache: "no-cache",
+                    credentials: "include",
+                    headers: {
+                        "Content-Type": "application/json",
+                    },
+                }
+            )
+                .then((resp) => {
+                    if (resp.status < 200 || resp.status > 299) {
+                        throw new Error(
+                            `${resp.status} ${resp.statusText} ${resp.text()}`
+                        );
+                    }
+                    return resp;
+                })
+                .then((resp) => resp.json())
+                .catch((e) => {
+                    console.log(e);
+                });
+
+            if (!data) {
+                data = {};
+            }
+
+            setData(data);
+        };
+        fetcher();
+    }, []);
+
+    const sections = [
+        [
+            constants.KIND_ANALYSES,
+            constants.SECTION_RECENT,
+            getMessage("recentAnalyses"),
+        ],
+        [
+            constants.KIND_ANALYSES,
+            constants.SECTION_RUNNING,
+            getMessage("runningAnalyses"),
+        ],
+        [
+            constants.KIND_APPS,
+            constants.SECTION_RECENTLY_ADDED,
+            getMessage("recentlyAddedApps"),
+        ],
+        [
+            constants.KIND_APPS,
+            constants.SECTION_PUBLIC,
+            getMessage("publicApps"),
+        ],
+        [constants.KIND_FEEDS, constants.SECTION_NEWS, getMessage("newsFeed")],
+        [
+            constants.KIND_FEEDS,
+            constants.SECTION_EVENTS,
+            getMessage("eventsFeed"),
+        ],
+    ];
+
+    return (
+        <div className={classes.gridRoot}>
```

Comment:
Could we add a static ID here so that if nothing gets returned from the service, the tests still have some known ID to check that the dashboard came up?

Also, a [loading skeleton](https://material-ui.com/components/skeleton/) might be good since I believe this is the default page for Sonora.

Reply:
I've added a loading skeleton, along with switching over to react-query.